src/Raven.Server/Documents/Queries/AST/QueryVisitor.cs

```diff
@@ -9,6 +10,8 @@ public abstract class QueryVisitor
 
         public void Visit(Query q)
         {
+            RuntimeHelpers.EnsureSufficientExecutionStack();
```

Comment:
I wonder if we could do better here by using `TryEnsureSufficientExecutionStack ` and throw own exception which would warn about huge number of clauses instead of a bit scary `InsufficientExecutionStackException`

Reply:
Oh I see. That's fine by me then